DFC.ServiceTaxonomy.GraphSync/Controllers/HomeController.cs

```diff
@@ -1,9 +1,48 @@
-﻿using Microsoft.AspNetCore.Mvc;
+﻿using System;
+using System.IO;
+using System.Text;
+using DFC.ServiceTaxonomy.GraphSync.Services.Interface;
+using Microsoft.AspNetCore.Mvc;
 
 namespace DFC.ServiceTaxonomy.GraphSync.Controllers
 {
     public class HomeController : Controller
     {
+        private readonly ISynonymService _synonymService;
+
+        public HomeController(ISynonymService synonymService)
+        {
+            _synonymService = synonymService ?? throw new ArgumentNullException(nameof(synonymService));
+        }
+
+        [HttpGet("GraphSync/Synonyms/{node}/{filename}")]
+        public IActionResult Synonyms(string node, string filename)
+        {
+            var synonyms = _synonymService.GetSynonyms(node);
+
+            var sb = new StringBuilder();
+            foreach (var item in synonyms)
+            {
+                sb.AppendLine(item);
+            }
+
+            var stream = new MemoryStream(buffer: Encoding.UTF8.GetBytes(sb.ToString()));
+
+            try
+            {
+                stream.Position = 0;
+                return new FileStreamResult(stream, "text/plain")
+                {
+                    FileDownloadName = filename
+                };
+            }
+            catch
```

Comment:
Can't you wrap the stream with an using (or use finally)?

Reply:
FileStreamResult disposes the stream passed in within it's internal method. If you wrap it you can get an already disposed exception. If you'd like us to handle the dispose I can change it to a FileResult instead.

https://stackoverflow.com/questions/10418594/with-filestreamresult-how-is-the-memorystream-closed